airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java

```diff
@@ -0,0 +1,183 @@
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.destination.mongodb.exception.MongodbDataBaseException;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.DestinationSyncMode;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MongodbDestination extends BaseConnector implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongodbDestination.class);
+
+  private static final String MONGO_URI = "mongodb://%s:%s@%s:%s/?authSource=%s&authMechanism=SCRAM-SHA-1";
+  private static final String HOST = "host";
+  private static final String PORT = "port";
+  private static final String DATABASE = "database";
+  private static final String AUTH_TYPE = "auth_type";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final String AUTHORIZATION = "authorization";
+  private static final String LOGIN_AND_PASSWORD = "login/password";
+  private static final String AIRBYTE_DATA = "_airbyte_data";
+
+  private final MongoDBNameTransformer namingResolver;
+
+  public MongodbDestination() {
+    namingResolver = new MongoDBNameTransformer();
+  }
+
+  public static void main(String[] args) throws Exception {
+    final Destination destination = new MongodbDestination();
+    LOGGER.info("starting destination: {}", MongodbDestination.class);
+    new IntegrationRunner(destination).run(args);
+    LOGGER.info("completed destination: {}", MongodbDestination.class);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      var uri = getMongoClientURI(config);
+      var client = getMongoClient(config.get(HOST).asText(), config.get(PORT).asInt(), uri);
+      var database = getMongoDatabase(client, config.get(DATABASE).asText());
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+    } catch (RuntimeException e) {
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.FAILED)
```

Comment:
Please add logger also here. In case something went wrong (like we already had several times previously) at least logs would show the reason. Also try to provide as many extended logs, as it's possible to a particular case. Thanks

Reply:
added new logger for check method